### DISCUSSION TITLE
kdocker: cannot detect spotify window

### DISCUSSION THREAD

**Discussion by realnc** [netblue30/firejail#5423] [Q&A] [answered]
I use kdocker to put spotify in the systray (unfortunately, only the windows version has support for the system tray.) I use this command:

```none
kdocker firejail -- /opt/spotify/spotify-client/spotify
```

But kdocker is unable to detect the Spotify window. Is there a way to make this work?

**Comment by rusty-snake** [comment]
How does kdocker work?

Does it work w/o firejail?

You can try `firejail --profile=spotify kdocker /opt/spotify/spotify-client/spotify`. Likely you need to modify spotify.profile to allow kdocker to work inside the sandbox.

**Reply by realnc** [reply]
Didn't work.

In its basic mode, kdocker waits for you to click on a window, and then "docks" it. Meaning it creates a systray icon, and when you click that icon, the application's window is hidden/shown.

Looking at the source code, it seems to be using Xlib (it makes calls to `XGetWindowAttributes()`, `XInternAtom()`, `XSendEvent()`, etc.)

**Comment by realnc** [comment, marked as answer]
Sorry, user error. It does work, but when using `kdocker firejail -- spotify`, kdocker looks for a window named "firejail". Duh!

One has to use `kdocker -n Spotify firejail -- spotify` instead.
